File: 03102023/01_08_Reduce_But_Grow/python/reduce_but_grow.py
# https://www.codewars.com/kata/57f780909f7e8e3183000078
# Level 8 kyu

# Directions:
# Given a non-empty array of integers, return the result of multiplying the values together in order. Example:
# [1, 2, 3, 4] => 1 * 2 * 3 * 4 = 24

# Function
def grow(numbers_list: list) -> int:
    answer: int = 1
    for number in numbers_list:
        answer *= number
    return answer


# Test Cases
# import codewars_test as test
# from solution import grow

# @test.describe("Fixed Tests")
# def fixed_tests():
#     @test.it('Basic Test Cases')
#     def basic_test_cases():
#         tests = [
#             [6 , [1, 2, 3]],
#             [16, [4, 1, 1, 1, 4]],
#             [64, [2, 2, 2, 2, 2, 2]],
#         ]
        
#         for exp, inp in tests:
#             test.assert_equals(grow(inp), exp)

File: 03102023/01_08_Reduce_But_Grow/java/ReduceButGrow.java
// https://www.codewars.com/kata/57f780909f7e8e3183000078
// Level 8 kyu

// Directions:
// Given a non-empty array of integers, return the result of multiplying the values together in order. Example:
// [1, 2, 3, 4] => 1 * 2 * 3 * 4 = 24

// Function
public class ReduceButGrow {

    public static void main(String[] args) {
        int[] integers = {1, 2, 3, 4};
        System.out.println(multiplyIntegers(integers));
    }

    public static int multiplyIntegers(int[] integers) {
        int answer = 1;
        for (int i : integers) {
            answer *= i;
        }
        return answer;
    }

}